subsys/net/lib/http/http_server_http1.c

```diff
@@ -90,37 +90,38 @@ static int handle_http1_static_resource(
 	int len;
 	int ret;
 
-	if (static_detail->common.bitmask_of_supported_http_methods & BIT(HTTP_GET)) {
-		data = static_detail->static_data;
-		len = static_detail->static_data_len;
-
-		if (static_detail->common.content_encoding != NULL &&
-		    static_detail->common.content_encoding[0] != '\0') {
-			snprintk(http_response, sizeof(http_response),
-				 RESPONSE_TEMPLATE "Content-Encoding: %s\r\n\r\n",
-				 "Content-Type: ",
-				 static_detail->common.content_type == NULL ?
-				 "text/html" : static_detail->common.content_type,
-				 len, static_detail->common.content_encoding);
-		} else {
-			snprintk(http_response, sizeof(http_response),
-				 RESPONSE_TEMPLATE "\r\n",
-				 "Content-Type: ",
-				 static_detail->common.content_type == NULL ?
-				 "text/html" : static_detail->common.content_type,
-				 len);
-		}
+	if (client->method != HTTP_GET) {
+		return send_http1_405(client);
+	}
 
-		ret = http_server_sendall(client, http_response,
-					  strlen(http_response));
-		if (ret < 0) {
-			return ret;
-		}
+	data = static_detail->static_data;
+	len = static_detail->static_data_len;
 
-		ret = http_server_sendall(client, data, len);
-		if (ret < 0) {
-			return ret;
-		}
+	if (static_detail->common.content_encoding != NULL &&
+	    static_detail->common.content_encoding[0] != '\0') {
+		snprintk(http_response, sizeof(http_response),
+			 RESPONSE_TEMPLATE "Content-Encoding: %s\r\n\r\n",
+			 "Content-Type: ",
+			 static_detail->common.content_type == NULL ?
+			 "text/html" : static_detail->common.content_type,
+			 len, static_detail->common.content_encoding);
+	} else {
+		snprintk(http_response, sizeof(http_response),
+			 RESPONSE_TEMPLATE "\r\n",
```

Comment:
The other branch above has header termination `\r\n\r\n` in it but this one does not, are we missing extra `\r\n` here?

Reply:
I think it's fine, note that the above branch glues one extra header (`Content-Encoding`) to the template, so it needs full `\r\n\r\rn` sequence to terminate headers. Here, one `\r\n` is already included in the template. Please note that I merge removed indentation for this code, it's not new one.

Reply:
Ok, I did not check what the template looks like, just wondered if there was an error.